src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +148,221 @@ async function uploadWithProgress({
+                getLogger().info('CreateBucket cancelled')
+                telemetry.recordS3CreateBucket({ result: 'Cancelled' })
+                return undefined
+            }
+
+            getLogger().info(`Creating bucket: ${bucketName}`)
+            try {
+                const bucket = await s3client.createBucket({ bucketName })
+
+                getLogger().info('Created bucket: %O', bucket)
+                window.showInformationMessage(localize('AWS.s3.createBucket.success', 'Created bucket: {0}', bucketName))
+                telemetry.recordS3CreateBucket({ result: 'Succeeded' })
+                return { Name: bucket.bucket.name }
+            } catch (e) {
+                getLogger().error(`Failed to create bucket ${bucketName}: %O`, e)
+                showErrorWithLogs(
+                    localize('AWS.s3.createBucket.error.general', 'Failed to create bucket: {0}', bucketName),
+                    Window.vscode()
+                )
+                telemetry.recordS3CreateBucket({ result: 'Failed' })
+                return undefined
+            }
+        }
+    } else {
+        return response.bucket
+    }
+    return undefined
+}
+
+export async function uploadFilePaletteCommand(
+    fileSizeBytes: FileSizeBytes = statFile,
+    window = Window.vscode(),
+    outputChannel = ext.outputChannel
+): Promise<void>{
+
+    //1 step: get the file that is currently opened in the editor
+    const activeEditor: TextEditor | undefined = vscode.window.activeTextEditor // returns an object of type TextEditor | undef 
+    const document = activeEditor?.document
+    let selectAnotherFile = false
+    let fileLocation: vscode.Uri | undefined
+
+
+    //2 step: prompt the quick picker so that user can specify if they want the
+    //current file or select other file from their system
+    //there is 2 possibilities
+    //there is an active editor with a file (document) open, then show the quickpick with the option 
+    //to browse for more files
+    //or there is no active editor, no file is open
+    if (!document) {
+        fileLocation = await promptForFileLocation(Window.vscode())
+    } 
+    else {
+        fileLocation = document.uri
+        const fileNameToDisplay = path.basename(fileLocation.fsPath)
+
+        //Now, ask the user with quickpicker to select file to use
+
+        //crete quick pick using picker.createQuickPick(options, items, buttons) that returns a QuickPick
+        const fileOption : vscode.QuickPickItem = { 
+            label: addCodiconToString('file', fileNameToDisplay) //TODO:: change parameters to correct ones
+        }
+        const selectMore: vscode.QuickPickItem = {
+            label: "Browse for more files..."
+        }
+
+        const picker = createQuickPick({
+            options: {
+                canPickMany: false,
+                ignoreFocusOut: true,
+                title: localize('AWS.message.selectFileUpload', 'Couldnt find it'), //not finding this wtf
```

Comment:
Localization might be broken, just best-effort keep this in sync with the nls file.

Reply:
Yeah localization does not currently work but it's best to always use `localize` on user-facing text to prevent possible tech-debt. It's pretty easy to remove all the localize calls but the reverse is not true.